src/main/java/com/onarandombox/MultiverseNetherPortals/listeners/MVNPEntityListener.java

```diff
@@ -49,41 +54,58 @@ public MVNPEntityListener(MultiverseNetherPortals plugin) {
         this.worldManager = this.plugin.getCore().getMVWorldManager();
         this.pt = new PermissionTools(this.plugin.getCore());
         this.playerErrors = new HashMap<String, Date>();
-        this.eventRecord = new HashMap<String, Location>();
         this.messaging = this.plugin.getCore().getMessaging();
         this.locationManipulation = this.plugin.getCore().getLocationManipulation();
-
+        this.eventRecord = new HashMap<>();
+        this.eventRecord.put(PortalType.ENDER, new HashSet<>());
+        this.eventRecord.put(PortalType.NETHER, new HashSet<>());
     }
 
-    protected void shootPlayer(Player p, Block block, PortalType type) {
+    protected boolean shootPlayer(Player p, Block block, PortalType type) {
         if (!plugin.isUsingBounceBack()) {
-            this.plugin.log(Level.FINEST, "You said not to use bounce back so the player is free to walk into portal!");
-            return;
+            this.plugin.log(Level.FINEST, "Bounceback is disabled, so the player is free to walk into portals!");
+            return false;
```

Comment:
A good change. I think even better might be `"Bounceback is disabled. The player is free to walk into the portal!"`

Reply:
Yeah, I agree.